opflexagent/vif_plug_vpp/vpp.py

```diff
@@ -0,0 +1,153 @@
+# Derived from nova/virt/libvirt/vif.py
+#
+# Copyright (C) 2018 Cisco Systems Inc.
+# Copyright 2011 OpenStack Foundation
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from os_vif import objects
+from os_vif import plugin
+from oslo_config import cfg
+
+from vif_plug_vpp import constants
+from vif_plug_vpp import exception
+from vif_plug_vpp import linux_net
+
+
+class VppPlugin(plugin.PluginBase):
+    """A VPP plugin that can setup VIFs in many ways
+
+    The VPP plugin supports VIF type VIFOpenVSwitch,
+    and will choose the appropriate plugging
+    action depending on the type of VIF config it receives.
+    """
+
+    NIC_NAME_LEN = 14
+
+    CONFIG_OPTS = (
```

Comment:
Is this used anywhere?

Reply:
yes in the same file.. function gen_port_name